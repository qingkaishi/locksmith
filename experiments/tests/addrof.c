#include <pthread.h>
#include <stdio.h>

int *j;
pthread_mutex_t l;

void * dostuff(void * a) {
  int k;
  
  pthread_mutex_lock(&l);
  j = &k;
  pthread_mutex_unlock(&l);
  return NULL;
}

int main() {
  pthread_t t1;
  pthread_mutex_init(&l, NULL);
  void* (*aa)(void*);
  aa = &dostuff;
  int i = 1;
  pthread_create(&t1, NULL, aa, NULL);
  pthread_create(&t1, NULL, &dostuff, NULL);
}


