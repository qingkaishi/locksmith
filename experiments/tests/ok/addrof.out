************************* STARTING *************************

shared: &j:addrof.c:4
&j:addrof.c:4 is protected by:
  concrete l:addrof.c:18
  l:addrof.c:5
*************************** DONE ***************************
